### PR TITLE
TST: Continue collecting arithmetic tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
This is split off of #22350, which may have been too ambitious.

tests/test_arithmetic.py is moved to tests/arithmetic/test_timedelta64.py.  Aside from being moved it is untouched.

### CODE REVIEW

**Review by gfyoung — APPROVED** [review]
cc @jreback

File: pandas/tests/arithmetic/test_numeric.py [hunk 1]
```diff
@@ -42,6 +42,30 @@ def test_operator_series_comparison_zerorank(self):
...
 
+    def test_df_numeric_cmp_dt64_raises(self):
+        # GH#8932, GH#22163
+        ts = pd.Timestamp.now()
+        df = pd.DataFrame({'x': range(5)})
+        with pytest.raises(TypeError):
+            df > ts
+        with pytest.raises(TypeError):
+            df < ts
+        with pytest.raises(TypeError):
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: shouldn't these be in test_datetime64?

File: pandas/tests/arithmetic/test_numeric.py [hunk 2]
```diff
@@ -42,6 +42,30 @@ def test_operator_series_comparison_zerorank(self):
...
+            ts < df
+        with pytest.raises(TypeError):
+            ts > df
+
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: can you add for tz as well (use the fixture)

File: pandas/tests/arithmetic/test_object.py [hunk 3]
```diff
@@ -180,3 +180,36 @@ def test_series_with_dtype_radd_timedelta(self, dtype):
 
         result = ser + pd.Timedelta('3 days')
         tm.assert_series_equal(result, expected)
+
```
Comment (jreback) [inline pandas/tests/arithmetic/test_object.py]: would move these to test_datetime64 (maybe rename that to test_datetime)

File: pandas/tests/arithmetic/test_numeric.py [hunk 4]
```diff
@@ -42,6 +42,30 @@ def test_operator_series_comparison_zerorank(self):
...
+        # GH#8058
+        # ops testing
+        a = pd.Series(np.random.randn(5), name=0)
+        b = pd.Series(np.random.randn(5))
+        b.name = pd.Timestamp('2000-01-01')
+        tm.assert_series_equal(a / b, 1 / (b / a))
+
 
 # ------------------------------------------------------------------
 # Numeric dtypes Arithmetic with Timedelta Scalar
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: are there timedelta tests that should be moved (ok for followup, just indicate)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arithmetic/test_numeric.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_numeric.py]: No, this is specifically testing that the numeric-dtypes are correctly handling the presence of datetime64-dtype, not vice-versa.

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_numeric.py]: Sure; follow-up.  This PR is pretty specifically only-moving.

File: pandas/tests/arithmetic/test_numeric.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_numeric.py]: Those should all be in test_arithmetic, which this PR moves to arithmetic.test_timedelta64

File: pandas/tests/arithmetic/test_object.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_object.py]: These are mixed-type; it's pretty specifically the object-dtyped Index/Series/Column that these tests are addressing.

**Comment by jbrockmendel** [comment]
Blocker for some parameterization/deduplication

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**